wgengine/netstack/netstack.go

```diff
@@ -377,11 +380,48 @@ func (ns *Impl) shouldProcessInbound(p *packet.Parsed, t *tstun.Wrapper) bool {
 	return false
 }
 
+var userPingSem = syncs.NewSemaphore(20) // 20 child ping processes at once
+
+func (ns *Impl) userPing(dstIP netaddr.IP, pingResPkt []byte) {
+	if !userPingSem.TryAcquire() {
+		return
+	}
+	defer userPingSem.Release()
+
+	t0 := time.Now()
+	var err error
+	switch runtime.GOOS {
+	case "windows":
+		err = exec.Command("ping", "-n", "1", "-w", "3000").Run()
```

Comment:
Where is destination ip?

Reply:
Whoops, I hadn't tested this on Windows yet 😛 

Fixed.